sources/org.osbuild.files

```diff
@@ -71,8 +83,21 @@ def main(options, checksums, cache, output):
 
     with concurrent.futures.ProcessPoolExecutor(max_workers=15) as executor:
         requested_urls = []
+        secrets = None
+
         for checksum in checksums:
             try:
+                if isinstance(urls[checksum], dict):
+                    secrets_module = urls[checksum].get("secrets")
+                    if secrets is None and secrets_module:
+                        secrets_module_name = secrets_module.get("name")
+                        # find the module location relative to the current file
+                        secrets_module_path = os.path.join(os.path.dirname(os.path.dirname(__file__)), "secrets", secrets_module_name)
+                        ret = subprocess.run(secrets_module_path, stdout=subprocess.PIPE)
+                        secrets = json.loads(ret.stdout)
```

Comment:
Sources are run on the host. Why not just include the code from your `secrets` module in here rather than calling out to it? What is the advantage of having this in `./secrets/`? Maybe @teg can elaborate?

Reply:
The idea was to make the certificate providers plugins so they are not rh specific.

Reply:
But why do they have to be plugins? Why not put a python module into `./osbuild/rhsm.py` and call it from wherever we need it? What is the advantage of putting it as an executable into `./secrets/`?

The `org.osbuild.files` source must be aware of the specific module, since it needs to fetch the correct arguments from it.

Reply:
The main argument I can think of is that it fits into the rest of the architecture where we don't assume any of the modules (liking that new word) are written in any specific programming language. In this case, if the subscription manager's D-Bus API gets better to allow us to query the key via D-Bus we could re-write the tool in rust (or C) without pulling in the dbus bindings for python.
I realize this is more of a theoretical argument ;D

Reply:
As much as I love theoretical arguments (and agree with this one), I would be perfectly fine with the approach @dvdhrm suggested, we can always refactor this in the future.

Reply:
I pushed an update implementing this as @dvdhrm has suggested.